docs/api-reference/webgl/copy-and-blit.md

```diff
@@ -0,0 +1,163 @@
+  * `height` - (*number*, default: framebuffer height) The height of the area to be copied,
+  * `format` - (*GLenum*, default: GL.RGBA) The format of the data.
+  * `type` - (*GLenum*, default: type of `buffer` or `UNSIGNED_BYTE`) The type of the data.
+
+  Target options
+  * `buffer` - (*Buffer*) Buffer object, into which data to be copied.
+  * `byteOffset` - (*number*, default: 0) Byte offset from which data should be copied into buffer.
+
+Notes:
+
+* Reading from floating point textures is dependent on an extension both in WebGL1 and WebGL2.
+* When supported, the `{format: GL.RGBA, type: GL.FLOAT, ...}` combination becomes valid for reading from a floating-point color buffer.
+
+This function makes calls to the following WebGL APIs:
+
+[`gl.readPixels`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/readPixels), [`gl.bindFramebuffer`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bindFramebuffer), [`gl.bindBuffer`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bindBuffer)
+
+
+### copyToTexture(opts: Object) : Texture
+
+Copies pixels from a `Framebuffer` or `Texture` object into the specified area of a two-dimensional texture image or cube-map texture image. (gl.copyTexImage2D, gl.copyTexSubImage2D and gl.copyTexSubImage3D wrapper)
+
+* opts
+
+  Source options
+  * `source` (`Texture` or `Framebuffer`) - If provided this object will be bound and data copied from it.
+  * `x` (`GLint`, optional, default: 0) - x coordinate of the lower left corner where to start copying.
+  * `y` (`GLint`, optional, default: 0) - y coordinate of the lower left corner where to start copying.
+
+  Target options
+  * `texture` (`Texture`, optional) - If provided this object will be bound and data copied into it, if not provided `target` must be set and data will be copied to currenlty bound `Texture` object.
+  * `target` (`GLenum`, optional) - Binding point where target texture is currently bound. Either `texture` or `target` must be specified.
+  * `xoffset` (`GLint`, optional) - X offset with in target texture.
+  * `yoffset` (`GLint`, optional) - Y offset with in target texture.
+  * `zoffset` (`GLint`, optional, WebGL2) - Z offset with in target texture, when using copying into 2D Array of 3D texture.
+  * `width` (`GLint`, optional, default: texture.width) - Width of the target texture.
+  * `height` (`GLint`, optional, default: texture.height) - Height of the target texture.
+
+Notes:
+* `xoffset`, `yoffset`, `zoffset` : when an offset is specified, it implies we are copying data into a sub region of the target texture and internally `gl.copyTexSubImage2D` or `gl.copyTexSubImage3D` are used based on the `target`, for these cases it is assumed that target texture has enough GPU memory already allocated. There is not such restriction when copying to entire texture (i.e. none of the above offsets are set).
```

Comment:
There is not such restriction => Clarify - the target texture is automatically reallocated in this case? Or data that doesn't fit is discarded?